### DISCUSSION TITLE
[Feature] Creating a unique numeric ID for name, displayname, author

### DISCUSSION THREAD

**Discussion by Amidweiz** [ikemen-engine/Ikemen-GO#860] [Feature Request] [answered]
I think it would be a good feature if name, displayname, and author could be converted into a unique numeric value. This would have the benefit of being able to generate a unique ID for the opponent unlike p2name, p3name, enemy,name, etc. which only returns boolean values. Not sure if this should have it's own triggers like p2NameID, p2AuthorID, etc. or just go off a simple enemy,nameID redirect.

Not sure if the value should be the ASCII or unicode decimal values of the string concatenated like

name = 78 + 41 + 77 + 69 = 265

or just have the values in series

name = 78417769

I think the former would be more compact that the ladder. A potential use case for this would be creating a unique profile for a self learning AI that can profile itself to a specific character but I'm sure other use cases would crop up. This would save the hassle of creating one from reading enemy,const(data._x_) values and concatenating them to create them. Not sure if command, and helpername should also be included as well or not.

**Comment by K4thos** [comment, marked as answer]
Can't say I like this idea. There is already this functionality that can be used to group characters:
https://github.com/K4thos/Ikemen_GO/wiki/Triggers#changed_const
Using numbers like this makes the code hard to read and I don't see any major adventage in this approach.
